<a href="https://colab.research.google.com/github/ltoosaint24/DS-Unit-1-Sprint-2-Statistics/blob/master/(Loveline_Toussaint)LS_DS_123_Confidence_Intervals_AssignmentII.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [1]:
#Import statements

import pandas as pd
import numpy as numy
import matplotlib.pyplot as plt
from scipy.stats import t
from scipy import stats

In [2]:
url =  'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'
coffee_imp = pd.read_csv(url)
coffee_imp

Drink  Caffeine
0       1     47.32
1       2     43.78
2       3     48.12
3       4     43.25
4       5     46.42
5       6     45.16
6       7     45.17
7       8     42.48
8       9     47.70
9      10     36.62
10     11     39.06
11     12     34.59
12     13     45.96
13     14     41.19
14     15     35.81
15     16     32.56
16     17     40.06
17     18     42.09
18     19     39.30
19     20     40.37
20     21     41.92
21     22     40.90
22     23     38.43
23     24     34.09
24     25     38.81
25     26     40.00
26     27     38.90
27     28     40.38
28     29     32.98
29     30     35.59
30     31     44.00
31     32     34.85
32     33     33.41
33     34     30.56
34     35     37.94
35     36     29.38
36     37     37.03
37     38     34.00
38     39     35.40
39     40     33.44
40     41     33.04
41     42     33.55
42     43     35.91
43     44     36.81
44     45     32.04
45     46     32.30
46     47     29.93
47     48     28.43
48     49     35.34
49     50     30.64

In [4]:
#Print the top 5 rows
coffe_headfive = coffee_imp.head(5)
coffe_headfive

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.  Summarize your results in a sentence or two.

In [5]:
#Calculations here
#the mean of the Top 5 row
coffe_headfive.mean()

Drink        3.000
Caffeine    45.778
dtype: float64

In [8]:
#Standard deviation;
coffe_std = coffe_headfive.std()
coffe_std

Drink       1.581139
Caffeine    2.159727
dtype: float64

In [9]:
#The sample count 
coffe_count = coffe_headfive.count()
coffe_count

Drink       5
Caffeine    5
dtype: int64

In [10]:
#The standard error for the top five sample:
coffe_headfive_SE = coffe_std/(coffe_count**(1/2))
coffe_headfive_SE

Drink       0.707107
Caffeine    0.965859
dtype: float64

Based on the various test statistics, it can be noted that on average of 3 drinks, and about 2 drinks on average are caffiene. The standard deviation explains the count between each drink within an average to be 1.6, and about 2.2 for caffienated drinks. There is possible standard error of 0.70, and 0.96 for the number of drinks and caffienated drinks.

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [14]:
from scipy.stats import t

#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.

#The degree of freedom is equal to n -1
#The confident interval was changed to 0.93
t_star = t.ppf(0.95,df=222)
print('t_star =', t_star)

t_star = 1.6517463586794643


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [ ]:
#Calculations of the Margin of error for a 95%
#The margin of error, similar to reported errors found within distribution boundaries,
#for example where data may carry different attribute to cause there to be differences within ratios.import mat

In [16]:
#Margin of error:
import math
z_critical = stats.norm.ppf(q = 0.95) 
#The stats.py provided the critical z-score
#sample standard deviation is applied within the margin of error
#it based on the z-score multiplied by the standard deviation divided by
# the square root of the length of the sample
margin_of_error = z_critical * (coffe_std/ math.sqrt(len(coffe_headfive)))
margin_of_error


Drink       1.163087
Caffeine    1.588697
dtype: float64

The margin of error for the sample of drinks is approximately 1.16. There is a margin of error of caffeine drinks, 1.59. The margin of error gives an idea of how far off our estimates might be (with 95% conficence). We're trying to supply a plausible range for our parameter of interest (the true average of drinks and caffiene drinks)


In [17]:
# Margin of error for body temperature
t_star * 0.035 

0.057811122553781256

In [18]:
#Margin of error for coffee sample:
coff_star = t.ppf(0.95 , df =4)
#t-score printed for coffe sample
print(coff_star)

2.13184678133629


In [21]:
#Margin of error 
margin_err_coffee = coff_star * 0.035 
margin_err_coffee

0.07461463734677015

There is different margin of error for the t-score provided of the drink/coffee sample. The margin error according to the t-score, is approximately 0.075.

###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [22]:
#confidence interval with formula:
caff_sampleMean = coffe_headfive['Caffeine'].mean()
#This is added and subtracted from the margin of error:
LowCI = caff_sampleMean - margin_err_coffee
#This provides the confidence interval for the lower qaudrants based on 
#the subtraction
LowCI

45.70338536265323

In [23]:
UpperCI = caff_sampleMean + margin_err_coffee
UpperCI
#This is the confidence interval within the upper levels of confidence
#because the margin error was added to the sample mean.

45.85261463734677

###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [ ]:
#Calculations here

In [ ]:
#Calculations
#Start by calculating the t-interval
#mean for Caffiene:
coffee_mean = coffe_headfive['Caffeine'].mean()
#Standard deviation for caffiene:
sd_caffiene = coffe_headfive['Caffeine'].std()
#Standard error for caffiene:
se_caffiene= sd_caffiene/(len(coffe_headfive['Caffeine'])**(1/2))
#The confidence intervals based on the t-interval function in python:
t_interval = t.interval( alpha =0.95, df = 4, loc =coffee_mean, scale = se_caffiene )
print(t_interval)

(43.09634494354886, 48.45965505645114)


###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

The two confidence intervals are within close proximity. However, the t-interval functions provides a confidence interval values that is a scale 2.0 less, and approximately 2.3 more. This justifies the margin of error that was being observed.

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

Based on the t-interval function, we can be 95% confidence that the mean caffienated drink will be between 43.10 and 48.5. There is a margin of error of approximately 0.07.

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [24]:
#Calculations
#90% confidence interval:
t.interval(alpha =0.90, df =4, loc = caff_sampleMean, scale = se_caffiene)

(43.71893616688226, 47.83706383311774)

The estimate is more accurate, than precise, because of lower confidence percentage. Based on the t-interval function, we can be 90% confident that caffience drinks will be between 43.71 and 47.84.

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [27]:
#Calculations
#99% confidence interval:
t.interval(alpha =0.99 ,df =4,  loc = caff_sampleMean, scale = se_caffiene)

(41.33109259700504, 50.22490740299496)

The results are considered to be more precise, because we can have a higher percentage of confidence. We can be 99% confidence that mean caffiene drink will be between 41.3 and 50.2. This mean that the statistics assures that there is higher probability of experiencing caffiene averages within those proportions.

## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


Yes, because the 34 mg fall below the qauntity of the low confidence level, and upper confidence levels. This significantly different because the large deviation from the two measures.

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

The increase samples provides a greater ratio to validate the conficence intervals. There are more data to increase accuracy, and results may be more precise if there larger frequency expressed within the data.

In [28]:
#Calculations here.
coff_starfifty = t.ppf( 0.95, df = 49) #t-stats for sample of 50 because of the increase of degree of freedom
coff_starfifty

1.6765508919142629

In [33]:
coff_starhund = t.ppf(0.95, df =99)
coff_starhund

1.6603911559963895

In [29]:
coff_star #t-stats for sample 5

2.13184678133629

In [ ]:
#The t-stats for the sample of 50 is less than the t-stats 
#of the sample of 5. This is the first unit of measure that indicate that 
#there will be a difference within margin of error

In [31]:
margine_or_errfifty = coff_starfifty * 0.035
margine_or_errfifty

0.0586792812169992

In [34]:
margine_orerrhund = coff_starhund * 0.035
margine_orerrhund

0.05811369045987364

The margin of error is less than the margin of error for the sample of 5 (this was approximately 0.07), as the sample size increased to 50 and 100. The larger sample had a margin of error within the close proximity (both were approximately 0.06).

Compare MEs here.

